src/gamecontrol.cpp

```diff
@@ -15,13 +15,128 @@
 #include "people.h"
 #include "window.h"
 #include "dates.h"
-#include "gamemode.h"
 #include "viewport.h"
 #include "weather.h"
 #include "freerct.h"
 
+GameModeManager _game_mode_mgr; ///< Game mode manager object.
+
+/** Runs various procedures that have to be done yearly. */
+void OnNewYear()
+{
+	// Nothing (yet) needed.
+}
+
+/** Runs various procedures that have to be done monthly. */
+void OnNewMonth()
+{
+	_finances_manager.AdvanceMonth();
+	_rides_manager.OnNewMonth();
+}
+
+/** Runs various procedures that have to be done daily. */
+void OnNewDay()
+{
+	_rides_manager.OnNewDay();
+	_guests.OnNewDay();
+	_weather.OnNewDay();
+	NotifyChange(WC_BOTTOM_TOOLBAR, ALL_WINDOWS_OF_TYPE, CHG_DISPLAY_OLD, 0);
+}
+
+/**
+ * For every frame do...
+ * @param frame_delay Number of milliseconds between two frames.
+*/
+void OnNewFrame(uint32 frame_delay)
+{
+	_window_manager.Tick();
+	_guests.DoTick();
+	DateOnTick();
+	_guests.OnAnimate(frame_delay);
+	_rides_manager.OnAnimate(frame_delay);
+}
+
+GameControl::GameControl()
+{
+	this->running = true;
+}
+
+GameControl::~GameControl()
+{
+}
+
+/**
+ * Run latest game control action.
+ * @pre next_action should not be equal to #GCA_NONE.
+ */
+void GameControl::RunAction()
+{
+	assert(this->next_action != GCA_NONE);
+
+	switch (this->next_action) {
+		case GCA_NEW_GAME:
+		case GCA_LOAD_GAME:
+			this->ShutdownLevel();
+
+			if (this->next_action == GCA_NEW_GAME) {
+				this->NewLevel();
+			} else {
+				LoadGameFile(this->fname);
+			}
+
+			this->StartLevel();
+			break;
+
+		case GCA_SAVE_GAME:
+			SaveGameFile(this->fname);
+			break;
+		
+		case GCA_QUIT:
+			this->ShutdownLevel();
+			this->running = false;
+			break;
+
+		default:
+			NOT_REACHED();
+	}
+
+	this->next_action = GCA_NONE;
+}
+
+/** Prepare for a #GCA_NEW_GAME action. */
+void GameControl::NewGame()
+{
+	this->next_action = GCA_NEW_GAME;
+}
+
+/**
+ * Prepare for a #GCA_LOAD_GAME action.
+ * @param fname Name of the file to load.
+ */
+void GameControl::LoadGame(const char *fname)
+{
+	this->fname = fname;
+	this->next_action = GCA_LOAD_GAME;
+}
+
+/**
+ * Prepare for a #GCA_SAVE_GAME action.
+ * @param fname Name of the file to write.
+ */
+void GameControl::SaveGame(const char *fname)
```

Comment:
Yikes, thanks for pointing this out, it appears I need to remember what language I'm using here...